### PR TITLE
Add dependency for building on the buildfarm

### PR DESCRIPTION
**Pull request by clalancette**
There are three changes in here:

1.  Add a direct Ogre dependency.  rviz_default_plugins definitely has a direct dependency on Ogre, so it should link in the correct things and export the dependency.
1.  Add a resource_retriever dependency for export.  This is because rviz_default_plugins again depends directly on it.
1.  Add a `target_link_libraries` on `resource_retriever::resource_retriever`.  In theory this shouldn't be necessary, but it won't link without it right now.  This is a workaround for now to get the buildfarm turning over.

### CODE REVIEW

File: rviz_default_plugins/CMakeLists.txt
Comment (wjwwood) [inline rviz_default_plugins/CMakeLists.txt:597]: This is likely an issue with resource_retriever's cmake no? It seems related to the changes @dirk-thomas has been making, but I don't have enough experience with those changes to say what might be the issue.

**Review by wjwwood — APPROVED** [review]

**Comment by wjwwood** [comment]
While adding a dependency on ogre may be the "right" thing to do here, I think there's underlying issue as to why it is no longer being transitively provided by rviz rendering, which again I can only assume has something to do with the changes @dirk-thomas has been making w.r.t. the new cmake style.

File: rviz_default_plugins/CMakeLists.txt
Comment (clalancette) [reply rviz_default_plugins/CMakeLists.txt:597]: Yeah, that very well could be.  I don't know why this isn't working here, and I also don't have enough experience.  These changes are enough to unblock the buildfarm (I think), but we should investigate why the transitive stuff doesn't seem to be working.

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10654/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6083/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8670/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10561/)

**Comment by clalancette** [comment]
I did some additional looking here.  At least *part* of the problem is that the version of ament_cmake that we currently have released is too old, and is missing https://github.com/ament/ament_cmake/commit/0fc4000815d5300d2b83bf0fddfd359624db074e .  I'll talk to @jacobperron about re-releasing that.

**Comment by jacobperron** [comment]
Latest changes for ament_cmake have been released: https://github.com/ros/rosdistro/pull/24766

**Comment by jacobperron** [comment]
It looks like the latest release of ament_cmake fixed the packaging job :tada: 
http://build.ros2.org/job/Fbin_uF64__rviz_default_plugins__ubuntu_focal_amd64__binary/8/

@clalancette Thanks for investigating!

**Comment by wjwwood** [comment]
So, I didn't read the last comment closely enough. I guess now we need to revert part of this and do another release, but I'll wait for feedback from @clalancette first.

**Comment by clalancette** [comment]
> So, I didn't read the last comment closely enough. I guess now we need to revert part of this and do another release, but I'll wait for feedback from @clalancette first.

Yeah, I think we don't need the explicit `target_link_libraries` to `resource_retriever` anymore.  I'm going to open a PR to revert just that bit.  The rest of it is correct, I think.

**Comment by clalancette** [comment]
> > 
> 
> Yeah, I think we don't need the explicit `target_link_libraries` to `resource_retriever` anymore. I'm going to open a PR to revert just that bit. The rest of it is correct, I think.

Nope, never mind.  I can't get it to compile without that explicit link.  I still don't know why that is, but I think we should move forward with this release and we can investigate it later.

**Comment by wjwwood** [comment]
I already released it yesterday, so if nothing can be reverted, then there's nothing new to release atm. 👍

**Pull request merged**
